### PR TITLE
Always create vendor/BUILD in hack/update-bazel.sh

### LINKED ISSUE #51075: update-bazel ignores vendor/BUILD
At this point I don't know if it is a valid file or not?

```
thockin@freakshow2 kubernetes json-iterator /$ rm vendor/BUILD 
thockin@freakshow2 kubernetes json-iterator /$ ./hack/update-bazel.sh 
wrote "BUILD.bazel"
thockin@freakshow2 kubernetes json-iterator /$ ls -l vendor/BUILD
ls: cannot access vendor/BUILD: No such file or directory
thockin@freakshow2 kubernetes json-iterator /$ git co -- vendor/BUILD
thockin@freakshow2 kubernetes json-iterator /$ ./hack/update-bazel.sh 
wrote "BUILD.bazel"
thockin@freakshow2 kubernetes json-iterator /$ ls -l vendor/BUILD
-rw-r----- 1 thockin eng 23888 Aug 21 20:28 vendor/BUILD
```

@ixdy @mikedanese @spxtr

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: makes sure there's always a `vendor/BUILD` file.
When updating `godeps`, the [instructions](https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md) say to recursively remove `vendor/`, which also removes the `vendor/BUILD` file. Unless you manually recreate this file, running `update-bazel.sh` would instead update the `all-srcs` rule in the root `BUILD.bazel` file, which is not desired. `gazelle` and `kazel` won't create `vendor/BUILD` on their own, since there are no go sources directly in `vendor/`. 

With this PR, we'll make sure that the `vendor/BUILD` file always exists, creating it if necessary.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #51075

**Release note**:

```release-note
NONE
```

/assign @spxtr @mikedanese 
cc @thockin

### CODE REVIEW

File: hack/update-bazel.sh [hunk 1]
```diff
@@ -28,5 +28,7 @@ rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 9ba3a24eeffafa3a794b9e7312103424e7da26e9
 kube::util::go_install_from_commit github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle 82483596ec203eb9c1849937636f4cbed83733eb
 
+[[ -f "${KUBE_ROOT}/vendor/BUILD" ]] || touch "${KUBE_ROOT}/vendor/BUILD"
```
Comment (spxtr) [inline hack/update-bazel.sh]: I think just `touch "${KUBE_ROOT}/vendor/BUILD"` will suffice.

**Comment by spxtr** [comment]
Also `api/`, `cmd/`, `pkg/`, `plugin/`, `test/`, and `third_party/`.

File: hack/update-bazel.sh (hunk 1 above)
Comment (ixdy) [inline hack/update-bazel.sh]: I wasn't sure whether `git` would care if the mtime updated, but I guess it doesn't.

**Comment by ixdy** [comment]
`vendor/BUILD` is the only one that is likely to get deleted in normal workflows, since the instructions for updating `godeps` say to remove `vendor/`. The other `BUILD` files are not likely to be deleted.

**Comment by spxtr** [comment]
I suppose.

/lgtm

**Comment by ixdy** [comment]
#51154 ended up deleting `vendor/BUILD`, so I've added a commit here to recreate it.

**Comment by thockin** [comment]
Why do we need to have an empty file in the first place?

**Comment by spxtr** [comment]
> Why do we need to have an empty file in the first place?

It will put the all-srcs filegroup in place if there is a BUILD file, which will keep this giant list out of the top-level BUILD file:

```
filegroup(
    name = "package-srcs",
    srcs = glob(["**"]),
    tags = ["automanaged"],
    visibility = ["//visibility:private"],
)

filegroup(
    name = "all-srcs",
    srcs = [
        ":package-srcs",
        "//vendor/bitbucket.org/bertimus9/systemstat:all-srcs",
        "//vendor/bitbucket.org/ww/goautoneg:all-srcs",
        "//vendor/cloud.google.com/go/compute/metadata:all-srcs",
        "//vendor/cloud.google.com/go/internal:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/arm/compute:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/arm/containerregistry:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/arm/disk:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/arm/network:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/arm/storage:all-srcs",
        "//vendor/github.com/Azure/azure-sdk-for-go/storage:all-srcs",
        "//vendor/github.com/Azure/go-ansiterm:all-srcs",
        "//vendor/github.com/Azure/go-autorest/autorest:all-srcs",
        "//vendor/github.com/MakeNowJust/heredoc:all-srcs",
        "//vendor/github.com/Microsoft/go-winio:all-srcs",
        "//vendor/github.com/NYTimes/gziphandler:all-srcs",
        "//vendor/github.com/PuerkitoBio/purell:all-srcs",
        "//vendor/github.com/PuerkitoBio/urlesc:all-srcs",
        ...
)
```

**Pull request merged**